GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs

```diff
@@ -99,5 +100,198 @@ internal override List<PhaseData> GetPhases(ParsedEvtcLog log, bool requirePhase
+                            attackEnd = Math.Min((int)stunSegment.Start, attackEnd); // Start of stun
+                        }
+                        Segment detSegment = target.GetBuffStatus(log, Determined762, c.Time, c.Time + duration).FirstOrDefault(x => x.Value > 0);
+                        if (detSegment != null)
+                        {
+                            attackEnd = Math.Min((int)detSegment.Start, attackEnd); // Start of determinated
+                        }
+
+                        replay.Decorations.Add(new DoughnutDecoration(true, -expectedHitTime, 0, 1500, (start, attackEnd), "rgba(255, 0, 0, 0.2)", new AgentConnector(target)));
+                        replay.Decorations.Add(new DoughnutDecoration(true, 0, 0, 1500, (start, attackEnd), "rgba(255, 0, 0, 0.2)", new AgentConnector(target)));
+                    }
+                    // Tail Swipe
+                    var tailLash = casts.Where(x => x.SkillId == TailLashSiax).ToList();
+                    foreach (AbstractCastEvent c in tailLash)
+                    {
+                        int duration = 1500;
+                        int openingAngle = 144;
+                        int radius = 600;
+                        replay.Decorations.Add(new FacingPieDecoration(((int)c.Time, (int)c.Time + duration), new AgentConnector(target), replay.PolledRotations, radius, openingAngle, "rgba(250, 120, 0, 0.2)"));
+                    }
+                    // 66% and 33% phases
+                    var causticExplosionPhases = casts.Where(x => x.SkillId == CausticExplosionSiaxPhase66 || x.SkillId == CausticExplosionSiaxPhase33).ToList();
+                    foreach (AbstractCastEvent c in causticExplosionPhases)
+                    {
+                        int duration = 20000;
+                        int start = (int)c.Time;
+                        int expectedHitTime = (int)c.Time + duration;
+                        int attackEnd = (int)c.Time + duration;
+
+                        Segment detSegment = target.GetBuffStatus(log, Determined762, c.Time, c.Time + duration).FirstOrDefault(x => x.Value > 0);
+                        if (detSegment != null)
+                        {
+                            attackEnd = Math.Min((int)detSegment.End, attackEnd); // End of determinated
+                        }
+
+                        replay.Decorations.Add(new CircleDecoration(true, expectedHitTime, 1500, (start, attackEnd), "rgba(255, 0, 0, 0.2)", new AgentConnector(target)));
+                        replay.Decorations.Add(new CircleDecoration(true, 0, 1500, (start, attackEnd), "rgba(255, 0, 0, 0.2)", new AgentConnector(target)));
+                    }
+                    // Poison AoE
+                    var vileSpit = casts.Where(x => x.SkillId == VileSpit).ToList();
+                    EffectGUIDEvent poisonField = log.CombatData.GetEffectGUIDEvent(EffectGUIDs.VileSpitSiax);
+                    if (vileSpit.Count > 0 && poisonField != null)
```

Comment:
Just for information, what is the reasoning behind the cout check?

Reply:
nothing specific, was a double measure to check that the skill got casted other than just the event, can be removed if you want

Reply:
Yeah, if they are not useful, they should be removed.